sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NumberPatternShouldBeRegular.cs

```diff
@@ -47,7 +47,7 @@ public void NonCompliantCases()
             int million = 1_000_00_000;  // Noncompliant {{Review this number; its irregular pattern indicates an error.}}
 //                        ^^^^^^^^^^^^
 
-            var x = 1_234.56_78_1; // Noncompliant
+            var x = 1_234.56_78_123; // Noncompliant
 
```

Comment:
do we have any tests with `____`? it would be nice to have something like `0xFF___FF___FF` - I believe it's Noncompliant due to line 144 in `NumberPatternShouldBeRegular.cs`

and I think it's fine to be noncompliant (even if the pattern is regular, but using multiple underscores should usually be a mistake)

Reply:
and also with `0xFF________FF___FF`

this program runs:
```
var x = 0xFF________FF___FF;
Console.WriteLine(x);
```

Reply:
Added the tests, and took the comment suggestions.